decoder.go

```diff
@@ -574,3 +587,12 @@ func (d *decoder) decodeStruct(name string, node ast.Node, result reflect.Value)
 
 	return nil
 }
+
+// findNodeType returns the type of ast.Node
+func findNodeType() reflect.Type {
+	var nodeContainer struct {
+		Node ast.Node
+	}
+	value := reflect.ValueOf(nodeContainer).FieldByName("Node")
+	return value.Type()
```

Comment:
I think this can be simplified to `var v ast.Node; return reflect.ValueOf(v).Type()`, correct?